src/tagging/tests/tagging.test.js

```diff
@@ -1,62 +1,104 @@
-  },
+    values: [
+      { description: 'Knedlik', id: 51 },
+      {
+        description:
+          'Daenerys Stormborn of the House Targaryen, First of Her Name,...and Mother of Dragons',
+        id: 52
+      }
+    ]
+  }
 ];
 
 const selectedTagCategory = { description: 'animal', id: 1 };
 const selectedTagValue = { description: 'duck', id: 1 };
-const assignedTags = [{ tagCategory: { description: 'Name', id: 1 }, tagValues: [{ description: 'Pepa', id: 11 }] }];
-
-const onChange = x => x;
-const onDelete = x => x;
+const assignedTags = [
+  {
+    description: 'Name',
+    id: 1,
+    values: [{ description: 'Pepa', id: 11 }]
+  }
+];
 
+const onChange = jest.fn();
+const onDelete = jest.fn();
 
 describe('Tagging component without redux mapping', () => {
   it('match snapshot', () => {
-    const tree = mount(<Tagging
-      tags={tags}
-      assignedTags={assignedTags}
-      onTagValueChange={onChange}
-      onTagCategoryChange={onChange}
-      onTagDeleteClick={onDelete}
-      selectedTagCategory={selectedTagCategory}
-      selectedTagValue={selectedTagValue}
-    />);
-    expect(toJson(tree)).toMatchSnapshot();
+    const component = mount(
+      <Tagging
+        tags={tags}
+        assignedTags={assignedTags}
+        onTagValueChange={onChange}
+        onTagMultiValueChange={onChange}
+        onTagCategoryChange={onChange}
+        onTagDeleteClick={onDelete}
+        selectedTagCategory={selectedTagCategory}
+        selectedTagValue={selectedTagValue}
+      />
+    );
+    const tree = toJson(component);
+    expect(tree).toMatchSnapshot();
   });
 
   it('should call methods', () => {
     const onTagCategoryChange = jest.fn();
     const onTagValueChange = jest.fn();
     const onTagDeleteClick = jest.fn();
-    const wrapper = shallow(<Tagging
-      tags={tags}
-      assignedTags={assignedTags}
-      onTagValueChange={onTagValueChange}
-      onTagCategoryChange={onTagCategoryChange}
-      onTagDeleteClick={onTagDeleteClick}
-      selectedTagCategory={selectedTagCategory}
-      selectedTagValue={selectedTagValue}
-    />);
+    const onTagMultiValueChange = jest.fn();
+    const wrapper = mount(
+      <Tagging
+        tags={tags}
+        assignedTags={assignedTags}
+        onTagValueChange={onTagValueChange}
+        onTagMultiValueChange={onTagMultiValueChange}
+        onTagCategoryChange={onTagCategoryChange}
+        onTagDeleteClick={onTagDeleteClick}
+        selectedTagCategory={selectedTagCategory}
+        selectedTagValue={selectedTagValue}
+      />
+    );
     wrapper.instance().onTagCategoryChange('xaxa');
-    expect(onTagCategoryChange.mock.calls.length).toEqual(1);
+    expect(onTagCategoryChange.mock.calls).toHaveLength(1);
     wrapper.instance().onTagValueChange('wawa');
-    expect(onTagCategoryChange.mock.calls.length).toEqual(1);
+    expect(onTagCategoryChange.mock.calls).toHaveLength(1);
```

Comment:
@PanSpagetka Are you testing that `onTagCategoryChange` does *not* get called when you call `onTagValueChange`?

Or is this a copypaste bug and the other 2 functions should also have 1 call?

Reply:
Such PR necromancy :D I don't really remember, but I think that it is testing that each function was called once. I am sorry, but I don't understand where is the problem?

Reply:
Your calling each of the 3 functions...

and then testing the first one was called only once, and testing it 3 times ;)